File: 1회차/programmers_algorithm/4.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int c, int y) {
    vector<int> answer; answer.resize(2,0);
    for (int a = 1; a*a <= y; a++) {
        if (y%a==0) {
            int b=y/a;
            if ((a+2)*(b+2)-a*b==c) {
                answer={max(a,b)+2,min(a,b)+2};
                break;
            }
        }
    }
    return answer;
}

File: 1회차/programmers_algorithm/4.py
import math
def solution(c, y):
    answer = []
    for a in range(1, int(math.sqrt(y))+1):
        if y%a==0:
            b=y/a
            if 2*(a+b)+4==c:
                answer.append(max(a,b)+2)
                answer.append(min(a,b)+2)
                break
    return answer